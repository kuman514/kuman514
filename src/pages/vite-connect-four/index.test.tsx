import { render } from '@testing-library/react';

import ViteConnectFourPage from '.';

describe('ViteConnectFourPage', () => {
  it('should show as expected', () => {
    const { container } = render(<ViteConnectFourPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should show as expected with edited date', () => {
    const { container } = render(
      <ViteConnectFourPage editedDate={new Date('2023-05-29')} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
