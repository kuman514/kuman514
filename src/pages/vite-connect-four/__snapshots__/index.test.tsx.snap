// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViteConnectFourPage should show as expected 1`] = `
<main
  class="min-h-screen flex flex-col w-full flex flex-col items-center"
>
  <div
    aria-label="banner-image"
    class="pt-16 w-full h-[50vh]"
  >
    <img
      alt="ViteConnectFour 배너 이미지"
      class="w-full h-full object-cover"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </div>
  <div
    class="flex flex-col px-8 py-24 gap-16 sm:px-24 max-w-6xl"
  >
    <h1
      class="text-bold text-4xl sm:text-5xl md:text-6xl"
    >
      Vite 커넥트 포 (ViteConnectFour)
    </h1>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        개요
      </h2>
      <section>
        <ul
          class="list-disc pl-8"
        >
          <li>
            React + Vite + Zustand를 이용하여 만든, 커넥트 포 게임.
          </li>
          <li>
            개인 프로젝트
          </li>
          <li>
            구현 내용 요약
            <ul
              class="list-disc pl-8"
            >
              <li>
                프로젝트 세팅, 최소 구현 조건 설정, 태스크 브레이크다운, 기능 구현, 테스트, 배포, 문서 및 업데이트 로그 작성, 유지보수를 담당함.
              </li>
              <li>
                커넥트 포 게임, 실행 취소 및 되돌리기 등등 각종 기능 구현.
              </li>
              <li>
                Vite + Yarn Berry로 프로젝트를 세팅하여 Create React App 대비 로딩 시간 30초 → 5초로 감소, 설치 속도가 5배 빨라짐.
              </li>
            </ul>
          </li>
        </ul>
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        방문하기
      </h2>
      <section>
        [
         
        <a
          class="text-blue-500 hover:text-purple-500 undefined"
          href="https://github.com/kuman514/ViteConnectFour"
          target="_blank"
        >
          GitHub 저장소
        </a>
         
        ] [
         
        <a
          class="text-blue-500 hover:text-purple-500 undefined"
          href="https://vite-connect-four.vercel.app/"
          target="_blank"
        >
          게임 플레이
        </a>
         
        ]
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        이 프로젝트를 만든 이유
      </h2>
      <section
        class="flex flex-row flex-wrap gap-2"
      >
        <p
          class="text-base font-normal indent-4 undefined"
        >
          가족들과 태블릿으로 커넥트 포 게임을 즐기고자 하는데, 가족이 쓰는 태블릿에는 커넥트 포 앱을 설치할 여유가 없었습니다. 그래서 가족들끼리 즐거움을 느끼면서도 태블릿의 저장용량도 아낄 수 있게 하고자, 웹 커넥트 포 앱을 직접 구현하게 되었습니다.
        </p>
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        플레이 방법 (Windows Chrome 기준)
      </h2>
      <section>
        <ul
          class="list-disc pl-8"
        >
          <li>
            플레이어 1이 빨강, 플레이어 2가 노랑.
          </li>
          <li>
            빨강이 먼저 돌을 놓으며, 이후 두 플레이어가 번갈아가며 돌을 놓습니다.
          </li>
          <li>
            돌은 낙하하기 때문에 맨 아래에서부터 쌓입니다.
          </li>
          <li>
            가로, 세로, 대각선 중 아무 한 방향으로 돌을 4개 이상 연결한 플레이어가 승리합니다.
          </li>
        </ul>
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        사용한 기술 스택
      </h2>
      <section
        class="flex flex-row flex-wrap gap-2"
      >
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          React
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          TypeScript
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          Vite
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          Yarn Berry
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          Zustand
        </span>
      </section>
    </section>
  </div>
</main>
`;

exports[`ViteConnectFourPage should show as expected with edited date 1`] = `
<main
  class="min-h-screen flex flex-col w-full flex flex-col items-center"
>
  <div
    aria-label="banner-image"
    class="pt-16 w-full h-[50vh]"
  >
    <img
      alt="ViteConnectFour 배너 이미지"
      class="w-full h-full object-cover"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </div>
  <div
    class="flex flex-col px-8 py-24 gap-16 sm:px-24 max-w-6xl"
  >
    <div
      class="w-full text-right"
    >
      최근 수정 일자: 
      2023
      년 
      5
      월
       
      29
      일
    </div>
    <h1
      class="text-bold text-4xl sm:text-5xl md:text-6xl"
    >
      Vite 커넥트 포 (ViteConnectFour)
    </h1>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        개요
      </h2>
      <section>
        <ul
          class="list-disc pl-8"
        >
          <li>
            React + Vite + Zustand를 이용하여 만든, 커넥트 포 게임.
          </li>
          <li>
            개인 프로젝트
          </li>
          <li>
            구현 내용 요약
            <ul
              class="list-disc pl-8"
            >
              <li>
                프로젝트 세팅, 최소 구현 조건 설정, 태스크 브레이크다운, 기능 구현, 테스트, 배포, 문서 및 업데이트 로그 작성, 유지보수를 담당함.
              </li>
              <li>
                커넥트 포 게임, 실행 취소 및 되돌리기 등등 각종 기능 구현.
              </li>
              <li>
                Vite + Yarn Berry로 프로젝트를 세팅하여 Create React App 대비 로딩 시간 30초 → 5초로 감소, 설치 속도가 5배 빨라짐.
              </li>
            </ul>
          </li>
        </ul>
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        방문하기
      </h2>
      <section>
        [
         
        <a
          class="text-blue-500 hover:text-purple-500 undefined"
          href="https://github.com/kuman514/ViteConnectFour"
          target="_blank"
        >
          GitHub 저장소
        </a>
         
        ] [
         
        <a
          class="text-blue-500 hover:text-purple-500 undefined"
          href="https://vite-connect-four.vercel.app/"
          target="_blank"
        >
          게임 플레이
        </a>
         
        ]
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        이 프로젝트를 만든 이유
      </h2>
      <section
        class="flex flex-row flex-wrap gap-2"
      >
        <p
          class="text-base font-normal indent-4 undefined"
        >
          가족들과 태블릿으로 커넥트 포 게임을 즐기고자 하는데, 가족이 쓰는 태블릿에는 커넥트 포 앱을 설치할 여유가 없었습니다. 그래서 가족들끼리 즐거움을 느끼면서도 태블릿의 저장용량도 아낄 수 있게 하고자, 웹 커넥트 포 앱을 직접 구현하게 되었습니다.
        </p>
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        플레이 방법 (Windows Chrome 기준)
      </h2>
      <section>
        <ul
          class="list-disc pl-8"
        >
          <li>
            플레이어 1이 빨강, 플레이어 2가 노랑.
          </li>
          <li>
            빨강이 먼저 돌을 놓으며, 이후 두 플레이어가 번갈아가며 돌을 놓습니다.
          </li>
          <li>
            돌은 낙하하기 때문에 맨 아래에서부터 쌓입니다.
          </li>
          <li>
            가로, 세로, 대각선 중 아무 한 방향으로 돌을 4개 이상 연결한 플레이어가 승리합니다.
          </li>
        </ul>
      </section>
    </section>
    <section
      class="flex flex-col gap-8 undefined"
    >
      <h2
        class="text-bold text-2xl sm:text-3xl md:text-4xl"
      >
        사용한 기술 스택
      </h2>
      <section
        class="flex flex-row flex-wrap gap-2"
      >
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          React
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          TypeScript
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          Vite
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          Yarn Berry
        </span>
        <span
          class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded-sm undefined"
        >
          Zustand
        </span>
      </section>
    </section>
  </div>
</main>
`;
